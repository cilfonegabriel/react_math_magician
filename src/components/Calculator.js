import { useState } from 'react';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [state, setState] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const handleClick = (e) => {
    setState((state) => calculate(state, e.target.textContent));
  };

  const { total, next, operation } = state;
  return (
    <section className="containerCalculator">
      <div className="pMath"> Let&apos;s Do Some Math </div>
      <div className="calculatorDiv">
        <div className="result">
          {total}
          {operation}
          {next}
        </div>
        <div className="buttons-calculator">
          <button onClick={handleClick} type="button" className="btn-calculator">AC</button>
          <button onClick={handleClick} type="button" className="btn-calculator">+/-</button>
          <button onClick={handleClick} type="button" className="btn-calculator">%</button>
          <button onClick={handleClick} type="button" className="btn-calculator color-right">รท</button>
          <button onClick={handleClick} type="button" className="btn-calculator">7</button>
          <button onClick={handleClick} type="button" className="btn-calculator">8</button>
          <button onClick={handleClick} type="button" className="btn-calculator">9</button>
          <button onClick={handleClick} type="button" className="btn-calculator color-right">x</button>
          <button onClick={handleClick} type="button" className="btn-calculator">4</button>
          <button onClick={handleClick} type="button" className="btn-calculator">5</button>
          <button onClick={handleClick} type="button" className="btn-calculator">6</button>
          <button onClick={handleClick} type="button" className="btn-calculator color-right">-</button>
          <button onClick={handleClick} type="button" className="btn-calculator">1</button>
          <button onClick={handleClick} type="button" className="btn-calculator">2</button>
          <button onClick={handleClick} type="button" className="btn-calculator">3</button>
          <button onClick={handleClick} type="button" className="btn-calculator color-right">+</button>
          <button onClick={handleClick} type="button" className="btn-calculator zero">0</button>
          <button onClick={handleClick} type="button" className="btn-calculator">.</button>
          <button onClick={handleClick} type="button" className="btn-calculator color-right">=</button>
        </div>
      </div>
    </section>
  );
};

export default Calculator;
