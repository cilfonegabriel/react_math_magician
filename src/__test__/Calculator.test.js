import React from 'react';
import { createRoot } from 'react-dom/client';
import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

it('no crashes when rendering', () => {
  const div = document.createElement('div');
  const root = createRoot(div);
  root.render(<Calculator />);
});

test('Home snapshot', () => {
  const tree = renderer.create(<Calculator />);
  expect(tree).toMatchSnapshot();
});
