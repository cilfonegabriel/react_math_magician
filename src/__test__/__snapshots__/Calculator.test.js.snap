// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home snapshot 1`] = `
<section
  className="containerCalculator"
>
  <div
    className="pMath"
  >
     Let's Do Some Math 
  </div>
  <div
    className="calculatorDiv"
  >
    <div
      className="result"
    />
    <div
      className="buttons-calculator"
    >
      <button
        className="btn-calculator"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="btn-calculator"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="btn-calculator"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="btn-calculator color-right"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        className="btn-calculator"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="btn-calculator"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="btn-calculator"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="btn-calculator color-right"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="btn-calculator"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="btn-calculator"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="btn-calculator"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="btn-calculator color-right"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="btn-calculator"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="btn-calculator"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="btn-calculator"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="btn-calculator color-right"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="btn-calculator zero"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="btn-calculator"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="btn-calculator color-right"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</section>
`;
